EIPS/eip-5719.md

```diff
@@ -0,0 +1,94 @@
+---
+eip: 5719
+title: Signature replacement interface
+description: Non-interactive replacing of smart contract wallet signatures that became stale due to configuration changes.
+author: Agustin Aguilar (@Agusx1211)
+discussions-to: https://ethereum-magicians.org/t/erc-signature-replacing-for-smart-contract-wallets/11059
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-26
+requires: 1271
+---
+
+## Abstract
+
+Smart contract wallet signed messages can become stale, meaning a signature that once was valid could become invalid at any point.
+
+Signatures MAY become stale for reasons like:
+
+* The internal set of signers changed
+* The wallet makes signatures expirable
+* The contract was updated to a new implementation
+
+The following standard allows smart contract wallets to expose a URI that clients can use to replace a stale signature with a valid one.
+
+## Motivation
+
+In contrast to EOA signatures, [EIP-1271](./eip-1271.md) signatures are not necessarily idempotent; they can become invalid at any point in time. This poses a challenge to protocols that rely on signatures remaining valid for extended periods of time.
+
+A signature MAY need to be mutated due to one of the following scenarios:
+
+1. The wallet removes a signer that contributed to signing the initial message.
+2. The wallet uses a Merkle tree to store signers, adding a new signer.
+3. The wallet uses a Merkle tree to store signatures, adding new signatures.
+4. The wallet is updated to a new implementation, and the signature schema changes.
+
+Non-interactive signature replacement SHOULD be possible, since the wallet that originally signed the message MAY NOT be available when the signature needs to be validated. An example use-case is the settlement of a trade in an exchange that uses an off-chain order book.
+
+## Specification
+
+The wallet contract MUST implement the following function:
+
+```solidity
+function getAlternativeSignature(bytes32 _digest) external view returns (string);
```

Comment:
I must be fundamentally missing something reading EIP-5719. 
What type of signature do you refer to? if it's ECDSA, what was the reason we are returning type `string`?